Gemfile

```diff
@@ -34,3 +34,5 @@ elsif RUBY_VERSION >= '1.9'
 else
   gem 'ruby-debug', :platforms => :mri
 end
+
+gem "psych", "< 4"
```

Comment:
I wish this got proper fix.

Reply:
Be my guest.

Reply:
Probably something like:

~~~
$ git diff
diff --git a/lib/mail/message.rb b/lib/mail/message.rb
index f5176ac8..95dfb7fe 100644
--- a/lib/mail/message.rb
+++ b/lib/mail/message.rb
@@ -1841,7 +1841,7 @@ module Mail
     end
 
     def self.from_yaml(str)
-      hash = YAML.load(str)
+      hash = YAML.load(str, :permitted_classes => [Mail::Body, Mail::SMTP, Symbol])
       m = self.new(:headers => hash['headers'])
       hash.delete('headers')
       hash.each do |k,v|
diff --git a/spec/mail/message_spec.rb b/spec/mail/message_spec.rb
index 033edf2b..0bdfbef6 100644
--- a/spec/mail/message_spec.rb
+++ b/spec/mail/message_spec.rb
@@ -196,7 +196,7 @@ describe Mail::Message do
 
       it "should serialize the basic information to YAML" do
         yaml = @yaml_mail.to_yaml
-        yaml_output = YAML.load(yaml)
+        yaml_output = YAML.respond_to?(:unsafe_load) ? YAML.unsafe_load(yaml) : YAML.load(yaml)
         expect(yaml_output['headers']['To']).to       eq "someone@somewhere.com"
         expect(yaml_output['headers']['Cc']).to       eq "someoneelse@somewhere.com"
         expect(yaml_output['headers']['Subject']).to  eq "subject"
@@ -214,7 +214,7 @@ describe Mail::Message do
       it "should serialize a Message with a custom delivery_handler" do
         @yaml_mail.delivery_handler = DeliveryAgent
         yaml = @yaml_mail.to_yaml
-        yaml_output = YAML.load(yaml)
+        yaml_output = YAML.respond_to?(:unsafe_load) ? YAML.unsafe_load(yaml) : YAML.load(yaml)
         expect(yaml_output['delivery_handler']).to eq "DeliveryAgent"
       end
 
@@ -226,7 +226,7 @@ describe Mail::Message do
 
       it "should not deserialize a delivery_handler that does not exist" do
         yaml = @yaml_mail.to_yaml
-        yaml_hash = YAML.load(yaml)
+        yaml_hash = YAML.respond_to?(:unsafe_load) ? YAML.unsafe_load(yaml) : YAML.load(yaml)
         yaml_hash['delivery_handler'] = "NotARealClass"
         deserialized = Mail::Message.from_yaml(yaml_hash.to_yaml)
         expect(deserialized.delivery_handler).to be_nil
@@ -234,7 +234,7 @@ describe Mail::Message do
 
       it "should deserialize parts as an instance of Mail::PartsList" do
         yaml = @yaml_mail.to_yaml
-        yaml_hash = YAML.load(yaml)
+        yaml_hash = YAML.respond_to?(:unsafe_load) ? YAML.unsafe_load(yaml) : YAML.load(yaml)
         deserialized = Mail::Message.from_yaml(yaml_hash.to_yaml)
         expect(deserialized.parts).to be_kind_of(Mail::PartsList)
       end
~~~

While this works running just the message_spec.rb,  I'm observing the following error when running the complete test suite:

~~~
  1) Mail::Message initialization YAML serialization should handle multipart mail
     Failure/Error: hash = YAML.load(str, :permitted_classes => [Mail::Body, Mail::SMTP, Symbol])
     
     Psych::DisallowedClass:
       Tried to load unspecified class: MyTestDeliveryMethod
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:99:in `find'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:28:in `load'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:424:in `resolve_class'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:213:in `visit_Psych_Nodes_Mapping'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/visitor.rb:30:in `visit'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/visitor.rb:6:in `accept'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:35:in `accept'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:343:in `each'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:343:in `each_slice'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:343:in `revive_hash'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/visitor.rb:30:in `visit'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/visitor.rb:6:in `accept'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:35:in `accept'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:318:in `visit_Psych_Nodes_Document'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/visitor.rb:30:in `visit'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/visitor.rb:6:in `accept'
     # /usr/share/gems/gems/psych-4.0.3/lib/psych/visitors/to_ruby.rb:35:in `accept'
     # ./lib/mail/message.rb:1870:in `from_yaml'
     # ./lib/mail/message.rb:1883:in `block (2 levels) in from_yaml'
     # ./lib/mail/message.rb:1883:in `map'
     # ./lib/mail/message.rb:1883:in `block in from_yaml'
     # ./lib/mail/message.rb:1873:in `each'
     # ./lib/mail/message.rb:1873:in `from_yaml'
     # ./spec/mail/message_spec.rb:240:in `block (4 levels) in <top (required)>'
~~~

Undefining the `MyTestDeliveryMethod` somewhere would be probably the right thing to do.

Reply:
I'm on it.

Reply:
Ideally it should be part of a PR that gets the base CI green (that was the point of restricting the version in this branch). But I think it's fine to do it here, since two unattended PRs seems enough already 😅, so feel free to send a PR to my fork 👍.

Reply:
Oh, I see that you already created #1476, then it's fine, if that one gets attended & merged first, I'll rebase this one and remove the upper bound from the Gemfile.